src/core/Env.js

```diff
@@ -319,6 +319,12 @@ getJasmineRequireObj().Env = function(j$) {
       }
     };
 
+    var ensureIsFunctionOrAsync = function(fn, caller) {
```

Comment:
I'm not completely wild about the naming here or in base.js, but I think that most Jasmine code that asks "is this thing a function?" isn't prepared to handle async functions. So it's probably best for j$.isFunction and ensureIsFunction to continue to reject async functions.